### DISCUSSION TITLE
ST-Link flashing

### DISCUSSION THREAD

**Discussion by KittenVillage** [TheKikGen/USBMidiKliK4x4#40] [Show and tell]
It is easier to flash via an ST-Link. I tried it the other way, very frustrating. Then I remembered I had gotten this graphic for another STM32 board I'd worked on. Stuck some female wire connecters into a pin header and then rested them against the USB housing to make a firm connection. 

https://imgur.com/a/HjXwX4g

I still had to bypass resistor 28, and then turn off read protection when I flashed the bootloader. Worked like a charm though and now I've got 16 midi devices showing up, looking forward to testing further. Thought I was going to throw this midi hub out yesterday. 

Feel free to include the images on the wiki here.

**Comment by TheKikGen** [comment]
This is the way to flash the MIDITECH 4x4 midi device with the USB MidiKlik 4x4 firmware.
